src/Sinks/Parquet/ParquetSink.cs

```diff
@@ -0,0 +1,279 @@
+                string.IsNullOrEmpty(schemaHash)
+                    ? $"part-{Guid.NewGuid()}-chunk.parquet"
+                    : $"part-{Guid.NewGuid()}-{schemaHash}-chunk.parquet");
+        }
+
+        private Task<UploadedBlob> SaveCompletionToken()
+        {
+            if (sink.dropCompletionToken)
+                // there seems to be an issue with Moq library and how it serializes BinaryData type
+                // in order to have consistent behaviour between units and actual runs we write byte 0 to the file
+                return sink.storageWriter.SaveBytesAsBlob(new BinaryData(new byte[] { 0 }), GetSavePath(),
+                    $"{schemaHash}.COMPLETED");
+
+            return Task.FromResult(new UploadedBlob());
+        }
+
+        private void WriteRowGroup(List<ParquetColumn> parquetColumns)
+        {
+            if (parquetColumns.Count == 0)
+            {
+                Log.Info("Received and empty chunk for {sinkPath}", sink.path);
+                return;
+            }
+
+            if (blockCount == 0) memoryStream = new MemoryStream();
+
+            writeInProgress = true;
+            blockCount += 1;
+            Log.Info("Processing inlet {blockCount} for {sinkPath}, size {dataLength}", blockCount, sink.path,
+                parquetColumns[0].Data.Length);
+            try
+            {
+                using (var parquetWriter = new ParquetWriter(sink.parquetSchema, memoryStream,
+                           append: sink.rowGroupsPerFile > 1 && blockCount > 1))
+                {
+                    using (var groupWriter = parquetWriter.CreateRowGroup())
+                    {
+                        foreach (var parquetColumn in parquetColumns) groupWriter.WriteColumn(parquetColumn);
+                    }
+                }
+
+                if (blockCount % sink.rowGroupsPerFile == 0 || IsClosed(sink.In))
+                {
+                    SavePart().ContinueWith(_ => GetAsyncCallback(PullOrComplete).Invoke());
+                    blockCount = 0;
+                }
+                else
+                {
+                    PullOrComplete();
+                }
+            }
+            catch (Exception ex)
+            {
+                switch (decider.Decide(ex))
+                {
+                    case Directive.Stop:
+                        taskCompletion.TrySetException(ex);
+                        FailStage(ex);
+                        break;
+                    case Directive.Resume:
+                        WriteRowGroup(parquetColumns);
+                        break;
+                    case Directive.Restart:
+                        PullOrComplete();
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+        }
+
+        private void CompleteSink()
+        {
+            taskCompletion.TrySetResult(NotUsed.Instance);
```

Comment:
this?

Reply:
Applied the correct code style to the project.